### PR TITLE
#17 - added Pytest to Travis with auto deploy

### PR DESCRIPTION
**Pull request by gurungrahul2**
Fixes #18 
added pytest and auto deploy script

### CODE REVIEW

**Comment by gurungrahul2** [comment]
@aashutoshrathi this is my first time with Travis :)

**Comment by aashutoshrathi** [comment]
Commands are right but pytest are failing.
The main task is to fix that.

**Comment by aashutoshrathi** [comment]
@gurungrahul2 Any updates buddy :smile:

**Comment by gurungrahul2** [comment]
@aashutoshrathi can you help me deploy my serverless

**Comment by aashutoshrathi** [comment]
Wait I think I found fix for this. @gurungrahul2

**Comment by gurungrahul2** [comment]
sure.

**Comment by aashutoshrathi** [comment]
Yes, I would love to help you. Ping me on Telegram

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,10 @@ install:
...
   - npm i -g npm
   - npm i -g serverless
+  - npm i serverless-python-requirements
+  - pip install pipenv
 # command to run tests
 script:
   # - sls deploy
-  - echo "Doing nothin now"
+  - pytest
+  - serverless deploy
```
Comment (aashutoshrathi) [inline .travis.yml]: Add this command before deploy
`serverless config credentials --provider aws --key $aws_access_key_id --secret $aws_secret_access_key --profile iam-aws`

Comment (gurungrahul2) [reply .travis.yml]: Sure

Comment (aashutoshrathi) [reply .travis.yml]: @gurungrahul2 on it?

Comment (gurungrahul2) [reply .travis.yml]: @aashutoshrathi done.

File: .travis.yml [hunk 2]
```diff
@@ -10,7 +10,11 @@ install:
...
   - npm i -g npm
   - npm i -g serverless
+  - npm i serverless-python-requirements
+  - pip install pipenv
 # command to run tests
 script:
   # - sls deploy
-  - echo "Doing nothin now"
+  - pytest
+  - serverless config credentials --provider aws --key $aws_access_key_id --secret $aws_secret_access_key --profile iam-aws
```
Comment (aashutoshrathi) [inline .travis.yml]: `serverless config credentials --provider aws --key ${aws_access_key_id} --secret ${aws_secret_access_key} --profile iam-aws`

Comment (gurungrahul2) [reply .travis.yml]: done

**Comment by aashutoshrathi** [comment]
Keys are added bro.

**Comment by aashutoshrathi** [comment]
Use this exact commands
```sh
serverless config credentials --provider aws --key ${aws_access_key_id} --secret ${aws_secret_access_key} --profile iam-aws
```

**Comment by gurungrahul2** [comment]
@aashutoshrathi earlier this command returned error in travis log, can you double check ?

**Comment by gurungrahul2** [comment]
branch rebased.

**Comment by gurungrahul2** [comment]
https://travis-ci.org/aashutoshrathi/git-profiler-bot/jobs/471763699

**Comment by aashutoshrathi** [comment]
Squash all commits and use this exact command.
`serverless config credentials --provider aws --key ${aws_access_key_id} --secret ${aws_secret_access_key} --profile iam-aws`

**Comment by gurungrahul2** [comment]
@aashutoshrathi why does my travis title have "No Environment Variable Set" tag?

**Comment by gurungrahul2** [comment]
https://docs.travis-ci.com/user/environment-variables/#defining-public-variables-in-travisyml

**Comment by gurungrahul2** [comment]
@aashutoshrathi commands are correct have a look at this https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions , I think after merging it will work, what are your thoughts on it ?

**Comment by gurungrahul2** [comment]
got to line 439 ,they still remoted the variables,https://travis-ci.org/aashutoshrathi/git-profiler-bot/jobs/471802980

**Comment by aashutoshrathi** [comment]
It deployed successfully.

**Pull request merged**